provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -680,13 +689,37 @@ func createClassicClusterObject(ctx context.Context,
 		builder.Version(vBuilder)
 	}
 
-	if state.AdminCredentials != nil {
-		htpasswdUsers := []*cmv1.HTPasswdUserBuilder{}
-		htpasswdUsers = append(htpasswdUsers, cmv1.NewHTPasswdUser().
-			Username(state.AdminCredentials.Username.ValueString()).Password(state.AdminCredentials.Password.ValueString()))
+	if common.HasValue(state.CreateAdminUser) && state.CreateAdminUser.ValueBool() ||
+		common.HasValue(state.AdminCredentials) {
+		if !common.HasValue(state.AdminCredentials) {
+			state.AdminCredentials = flattenAdminCredentials("", "")
+		}
+		username, password := expandAdminCredentials(ctx, state.AdminCredentials, diags)
+		if username == "" {
+			username = commonutils.ClusterAdminUsername
+		}
+		if password == "" {
+			password, err = idputils.GenerateRandomPassword()
+			if err != nil {
+				tflog.Error(ctx, "Failed to generate random password")
+				return nil, err
+			}
+		}
+
+		hashedPwd, err := idputils.GenerateHTPasswdCompatibleHash(password)
+		if err != nil {
+			tflog.Error(ctx, "Failed to hash the password")
+			return nil, err
+		}
+		htpasswdUsers := []*cmv1.HTPasswdUserBuilder{
+			cmv1.NewHTPasswdUser().Username(username).HashedPassword(hashedPwd),
+		}
 		htpassUserList := cmv1.NewHTPasswdUserList().Items(htpasswdUsers...)
 		htPasswdIDP := cmv1.NewHTPasswdIdentityProvider().Users(htpassUserList)
 		builder.Htpasswd(htPasswdIDP)
+		state.AdminCredentials = flattenAdminCredentials(username, password)
+	} else {
+		state.AdminCredentials = flattenAdminCredentials("", "")
 	}
```

Comment:
```suggestion
	}
	state.AdminCredentials = flattenAdminCredentials(username, password)
```
his is refactoring `refactoring-2`
See comment with refactoring `refactoring-1` for the whole suggestion

Reply:
fixed